### DISCUSSION TITLE
Serializing arrow-schema objects

### DISCUSSION THREAD

**Discussion by mlilback** [apache/arrow-rs#3771] [Q&A] [answered]
I'm trying to convert a Schema to json. All the structs I've looked at are annotated with
```
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```
However, there is no "serde" feature I can enable for the arrow crate. How do I enable this feature?

Here is the error I'm getting:
```
48   |             let json = serde_json::to_string(&schema).unwrap();
     |                        --------------------- ^^^^^^^ the trait `serde::ser::Serialize` is not implemented for `arrow_schema::schema::Schema`
     |                        |
     |                        required by a bound introduced by this call
```

I can see at that at one point there was no such cfg_attr, but looking through the history I can't find when or why this change was made.

**Comment by tustvold** [comment, marked as answer]
Arrow-schema has an optional `serde` dependency - https://crates.io/crates/arrow-schema/34.0.0/dependencies

If you add `features = ["serde"]` it should work

**Reply by mlilback** [reply]
I didn't realize you could directly import arrow-schema, I thought you had to import arrow. 

Thanks

**Reply by tustvold** [reply]
Yeah, see https://github.com/apache/arrow-rs/issues/2594
